### PR TITLE
Converted to swift 3.0 (Xcode 8 beta 2)

### PR DESCRIPTION
**Pull request by ldiqual**
@devxoul 

Fixes https://github.com/devxoul/JLToast/issues/84

All the migration was done using the Xcode migration code. Tests pass, demo app runs fine.

Let me know if you have any question.

### CODE REVIEW

**Comment by AlexMcArdle** [comment]
Code compiles and works with Carthage on Beta 2

**Comment by ldiqual** [comment]
Also confirmed that it compiles fine as a pod in my own app.

File: JLToast.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -450,6 +453,8 @@
 				PRODUCT_BUNDLE_IDENTIFIER = kr.xoul.JLToast;
 				PRODUCT_NAME = "$(TARGET_NAME)";
 				SKIP_INSTALL = YES;
+				SWIFT_OPTIMIZATION_LEVEL = "-Owholemodule";
```
Comment (devxoul) [inline JLToast.xcodeproj/project.pbxproj]: Is this line necessary? 🤔

File: JLToast/JLToast.swift [hunk 2]
```diff
@@ -85,8 +85,8 @@ public struct JLToastDelay {
     }
     
     override public func start() {
-        if !NSThread.isMainThread() {
-            dispatch_async(dispatch_get_main_queue(), {[weak self] in
+        if !Thread.isMainThread {
+            DispatchQueue.main.async(execute: {[weak self] in
```
Comment (devxoul) [inline JLToast/JLToast.swift]: This is really awesome :+1:

File: JLToast.xcodeproj/project.pbxproj (hunk 1 above)
Comment (AlexMcArdle) [inline JLToast.xcodeproj/project.pbxproj]: [useyourloaf.com/blog/swift-whole-module-optimization/](Here's a write-up on Whole Module Optimization.) Xcode 8 bugs you to turn it on when upgrading the project. For projects that are being built for release (i.e. libraries/frameworks) it is recommend to turn on.

**Comment by devxoul** [comment]
@ldiqual @AlexMcArdle Thanks for your contribution!

**Pull request merged**